team-validator.js

```diff
@@ -253,16 +253,15 @@ class Validator {
 				if (!set.moves[i]) continue;
 				let move = tools.getMove(Tools.getString(set.moves[i]));
 				if (!move.exists) return ['"' + move.name + '" is an invalid move.'];
-				set.moves[i] = move.name;
```

Comment:
It already is the guarantee, because the Pokemon ruleset requires it. I've added a changeset to enforce it there, since that's what it wants. (Alternatively the Pokemon ruleset could be changed to enforce names too, and then the validator changes wouldn't be needed.)